### PR TITLE
Feature/drop before win7 support in compiler (#6681)

### PR DESCRIPTION
**Pull request by dothebart**

### CODE REVIEW

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/2506/

File: CMakeLists.txt [hunk 1]
```diff
@@ -410,7 +410,7 @@ option(HOMEBREW
 if (WIN32)
   set(WINDOWS TRUE)
   set(MSBUILD TRUE)
-  add_definitions(-D_USE_MATH_DEFINES -DNOMINMAX)
+  add_definitions(-D_USE_MATH_DEFINES -DNOMINMAX -D_WIN32_WINNT=_WIN32_WINNT_VISTA -DINCL_EXTRA_HTON_FUNCTIONS -DPSAPI_VERSION=1)
```
Comment (Simran-B) [inline CMakeLists.txt]: Doesn't this target Vista / Windows Server 2008 and above in contrast to the PR title to drop support for OS before Windows 7?
https://docs.microsoft.com/en-us/cpp/porting/modifying-winver-and-win32-winnt?view=vs-2017

Comment (jsteemann) [reply CMakeLists.txt]: @dothebart: please clarify!

Comment (dothebart) [reply CMakeLists.txt]: We already discussed this before f2f - without these parameters only the latest windows targets are supported, with them the mentioned windows 7 will be too. The original commit message wasn't all the best - however there would also be parameters to support even older wintendos.

Comment (jsteemann) [reply CMakeLists.txt]: Does it mean what @simran-B mentioned does not apply here?

Comment (dothebart) [reply CMakeLists.txt]: well, regarding the quote "Windows 7 is lipstick on the Vista-Pig" its probably true.

**Pull request merged**